### PR TITLE
utils: move files that are not part of the module to scripts/

### PR DESCRIPTION
**Pull request by VincentVanlaer**
There are multiple scripts and software archives in utils that are not part of the module. This commit moves them to a separate folder, such that the utils folder only contains files related to the utils mesa module. The scripts that are part of the build are still in utils/, these should be removed once all modules have been migrated from to the new build system.

### CODE REVIEW

**Comment by earlbellinger** [comment]
Nice cleanup. But perhaps `$MESA_DIR/scripts` is a better home for these files?

**Comment by VincentVanlaer** [comment]
I missed the existence of the folder, that makes a lot of sense :smile:

**Comment by warrickball** [comment]
**tl;dr**: I'd say `makedepf90-2.8.8.tar.gz`, `ndiff-2.00.tar.gz` and `pgplot.tbz` aren't scripts, so putting them in a `scripts` folder doesn't really make sense, but I'm happy to move them until we decide what to do with them.

I can't offer a useful suggestion for where they *should* go but am also not sure if we need them. I'm not sure what `pgplot.tbz` is for but AFAIK `makedepf90` and `ndiff` are there to support non-SDK builds but I'm also not sure if:

* we bundle everything needed for non-SDK builds anyway (I'm thinking `crmath`/`crlibm`) or
* `ndiff` is actually used anywhere in the build anyway (probably became irrelevant after the bit-for-bit work).

It looks like `makedepf90` is (for the moment) available at [this GitHub mirror](https://github.com/outpaddling/makedepf90) and is available to install as a package on [Ubuntu](https://packages.ubuntu.com/noble/makedepf90) and [Fedora](https://packages.fedoraproject.org/pkgs/makedepf90/makedepf90/) (though unmaintained).

I'm happy to move the files in this PR and then I can propose to remove them, rather than muddying the water.

**Comment by fxt44** [comment]
pgplot.tbz is the unmodified package, https://sites.astro.caltech.edu/~tjp/pgplot/ . from it, pgstar was created.

**Comment by fxt44** [comment]
pgplot.tbz is the unmodified pgplot package - https://sites.astro.caltech.edu/~tjp/pgplot/ .  pgstar was built from it.

**Comment by rhdtownsend** [comment]
The pgplot (5.2) in the SDK is patched so that it supports libpng-1.5. 

> On Jul 30, 2024, at 4:43 PM, fxt ***@***.***> wrote:
> 
> 
> pgplot.tbz is the unmodified pgplot package - https://sites.astro.caltech.edu/~tjp/pgplot/ . pgstar was built from it.
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because your review was requested.Message ID: ***@***.***>

--
Rich Townsend • Professor of Astronomy
Astronomy Department • University of Wisconsin-Madison
Phone: 608-262-1752 • E-mail: ***@***.***

**Comment by fxt44** [comment]
in other words, pgplot.tbz could be deleted.

**Comment by rhdtownsend** [comment]
Along with ndiff, it seems to be redundant. Yes, pgplot and ndiff are required by non-SDK builds. But so is HDF5, crlibm, etc — it’s up to the non-SDK installer to take care that those deps are already built, right?

> On Aug 1, 2024, at 7:41 AM, fxt ***@***.***> wrote:
> 
> 
> in other words, pgplot.tbz could be deleted.
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because your review was requested.Message ID: ***@***.***>

--
Rich Townsend • Professor of Astronomy
Astronomy Department • University of Wisconsin-Madison
Phone: 608-262-1752 • E-mail: ***@***.***

**Comment by VincentVanlaer** [comment]
Similarly, there is lapack and blas source code in the mtx module. I have [a branch](https://github.com/VincentVanlaer/mesa/commit/f127627580a77ece3ae5ae0efa137671178c00ad) with that stuff removed. I can make a PR if that is the direction we want to move things.

**Review by pmocz — COMMENTED** [review]
Should we merge this and remove redundant files like pgplot.tbz, ndiff, etc in a next PR? Also ok to remove them at this time I think

**Comment by rhdtownsend** [comment]
> Should we merge this and remove redundant files like pgplot.tbz, ndiff, etc in a next PR? Also ok to remove them at this time I think

Thumbs up from me...
